examples/rust/ex1-volume/src/main.rs

```diff
@@ -0,0 +1,584 @@
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+//                             libCEED Example 1
+//
+// This example illustrates a simple usage of libCEED to compute the volume of a
+// 3D body using matrix-free application of a mass operator.  Arbitrary mesh and
+// solution orders in 1D, 2D and 3D are supported from the same code.
+//
+// The example has no dependencies, and is designed to be self-contained. For
+// additional examples that use external discretization libraries (MFEM, PETSc,
+// etc.) see the subdirectories in libceed/examples.
+//
+// All libCEED objects use a Ceed device object constructed based on a command
+// line argument (-ceed).
+
+use libceed::{prelude::*, Ceed};
+use structopt::StructOpt;
+
+// ----------------------------------------------------------------------------
+// Command line arguments
+// ----------------------------------------------------------------------------
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "libCEED Rust Example 1 - Volume",
+    about = "This example uses the mass matrix to compute the length, area, or volume of a region, depending upon runtime parameters."
+)]
+struct Opt {
+    /// Set libCEED backend resource to use
+    #[structopt(short = "c", long = "ceed", default_value = "/cpu/self")]
+    ceed_spec: String,
+    /// Set mesh dimension
+    #[structopt(short = "d", long = "dimension", default_value = "3")]
+    dim: usize,
+    /// Set polynomial degree for the mesh
+    #[structopt(short = "m", long = "mesh_degree", default_value = "4")]
+    mesh_degree: usize,
+    /// Set polynomial degree for the solution
+    #[structopt(short = "p", long = "solution_degree", default_value = "4")]
+    solution_degree: usize,
+    /// Set number of quadrature points in 1D
+    #[structopt(short = "q", long = "num_qpts", default_value = "6")]
+    num_qpts: usize,
+    /// Set approximate problem size
+    #[structopt(short = "s", long = "problem_size", default_value = "-1")]
+    problem_size_requested: i64,
+    /// Enable test mode
+    #[structopt(short = "t", long = "test")]
+    test: bool,
+    /// Use QFunctions from gallery
+    #[structopt(short = "g", long = "gallery")]
+    gallery: bool,
+}
+
+// ----------------------------------------------------------------------------
+// Example 1
+// ----------------------------------------------------------------------------
+fn main() {
+    let opt = Opt::from_args();
+    let _ = example_1(opt);
+}
+
+fn example_1(opt: Opt) -> Result<(), &'static str> {
```

Comment:
`&'static str` is limiting because you can't format output based on run-time state. I think `String` would make more sense here, or define an actual error type. Also note that `main` can return `Result` so you don't have to drop it `let _ = ` above.